### PR TITLE
Loading activity from json

### PR DESCRIPTION
**Pull request by gimait**
This is a first try on loading activity maps from JSON files.

I wanted to create the PR to ask for some feedback about the implementation, please let me know if I can do things in a nicer way.

Currently, no logic on the activities is done, only the mapping from JSON into different classes.

### CODE REVIEW

File: pycozmo/activity.py [hunk 1]
```diff
@@ -14,30 +16,389 @@
...
+
+        self.choice_type = choice_type
+        self.behaviors = behaviors
+        self.iteration = 0
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(
+            choice_type=data['type'],
+            behaviors=data['behaviors'] if 'behaviors' in data else []
```
Comment (zayfod) [inline pycozmo/activity.py:33]: Consider the following in such cases:
```
behaviors=data.get('behaviors', [])
```

File: pycozmo/activity.py [hunk 2]
```diff
@@ -14,30 +16,389 @@
...
+                out = self.behaviors[self.iteration]
+                self.iteration += 1
+                return out
+            else:
+                return None
+        elif self.choice_type == 'Scoring':
+            # TODO: add score based choice method
+            pass
+        else:
+            print(self.choice_type)
```
Comment (zayfod) [inline pycozmo/activity.py:53]: Raise an exception of some sort.

File: pycozmo/activity.py [hunk 3]
```diff
@@ -14,30 +16,389 @@
...
+        self.random_completions_needed_max = random_completions_needed_max
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(objective=data['objective'],
+                   behavior_ID=data['behaviorID'],
+                   ignore_if_locked=data['ignoreIfLocked'],
+                   probability_to_require_objective=data['probabilityToRequireObjective'],
+                   random_completions_needed_min=data['randomCompletionsNeededMin']
+                   if 'randomCompletionsNeededMin' in data else None,
```
Comment (zayfod) [inline pycozmo/activity.py:79]: If the default is 0, why pass None here?

Comment (zayfod) [reply pycozmo/activity.py:79]: This is the same as `random_completions_needed_min=data.get('randomCompletionsNeededMin')

File: pycozmo/activity.py [hunk 4]
```diff
@@ -14,30 +16,389 @@
...
+
+    elif info['activityType'] == 'NeedsExpression':
+        return NeedsActivity.from_json(info)
+
+    else:
+        return Activity(
+            activity_id=info['activityID'],
+            activity_type=info['activityType'],
+            strategy=info['activityStrategy']['type']
+        )
```
Comment (zayfod) [inline pycozmo/activity.py:369]: 👍

File: pycozmo/activity.py [hunk 5]
```diff
@@ -14,30 +16,389 @@
...
+
+
+def get_activity_files(resource_dir: str) -> List[str]:
+    activity_files = [resource_dir + '/cozmo_resources/config/engine/behaviorSystem/activities_config.json']
+    activity_folder = resource_dir + '/cozmo_resources/config/engine/behaviorSystem/activities/'
+    for root, dirs, files in os.walk(activity_folder):
```
Comment (zayfod) [inline pycozmo/activity.py:375]: As you are not using `dirs`, use `_` instead.

File: pycozmo/activity.py [hunk 6]
```diff
@@ -14,30 +16,389 @@
...
+    activity_files = get_activity_files(resource_dir)
+
+    for filename in activity_files:
+        with open(filename, 'r') as cf:
+            filtered_json = ''
+            for line in cf.readlines():
+                if '//' in line:
+                    wordlist = line.split(' ')
+                    # get all words before '//'
+                    line = ' '.join(wordlist[:wordlist.index('//')])
```
Comment (zayfod) [inline pycozmo/activity.py:395]: Uhh, they are using malformed JSON :/

**Review by zayfod — APPROVED** [review]
Looks good overall. You've decided to start with a hard one! 🍺

File: pycozmo/activity.py (hunk 1 above)
Comment (gimait) [reply pycozmo/activity.py:33]: This is so much nicer, thanks a lot!

File: pycozmo/activity.py (hunk 3 above)
Comment (gimait) [reply pycozmo/activity.py:79]: It's None due to copy-paste or autocomplete most likely. I'll correct it.

File: pycozmo/activity.py (hunk 4 above)
Comment (gimait) [reply pycozmo/activity.py:369]: There is one thing I wanted to ask. In this class, you added some slots. Should I include slots for the non-optional variables in the different activities?

File: pycozmo/activity.py (hunk 6 above)
Comment (gimait) [reply pycozmo/activity.py:395]: Yes, quite annoying haha I hope they are only using this format for the comments.

File: pycozmo/activity.py (hunk 4 above)
Comment (zayfod) [reply pycozmo/activity.py:361]: `__slots__` is a way to restrict what fields can be used in a class. Feel free to remove them for now for activities.

**Comment by zayfod** [comment]
Looks good. Merge as is?

**Comment by gimait** [comment]
Can be merged, yes, I'll move to another set of files.

**Pull request merged**